src/main/java/com/base64/d/StudyGuideChecker.java

```diff
@@ -8,127 +8,120 @@
-//                            .substring(0, matcher2.group(1).length() - 1);
-//                System.out.println(shorterStudyGuideLink);
-//                  links are good because regex replace all removes "\" if there.
-                    studyGuideLinks.add(shorterStudyGuideLink);
-
+                    // Some study guide links had '\' added for unknown reason.
+                    // This code checks for it and removes it before adding it to the studyGuide.
+                    String shorterStudyGuideLink = contentLinkMatcher.group(1).substring(0, contentLinkMatcher.group(1).length() - 1);
+
+                    if (contentLinkMatcher.group(1).substring(contentLinkMatcher.group(1).length() - 1).equals("\\")) {
+                        studyGuideLinks.add(shorterStudyGuideLink);
+                    } else {
+                        studyGuideLinks.add(contentLinkMatcher.group(1));
+                    }
                 }
 
                 System.out.println(studyGuideLinks.size() + " Unique Campus Community links found");
-                //remove all the false positives that are in JSON of study guide but not visible to front end user.
-                studyGuideLinks.removeAll(KnownFalsePositives.BLACK_LIST);
+                // Remove all the false positives that are in JSON of study guide but not visible to front end user.
+                studyGuideLinks.removeAll(KnownFalsePositives.WHITE_LIST);
                 System.out.println(studyGuideLinks);
                 totalLinksFound += studyGuideLinks.size();
 
             } catch (IOException e) {
                 e.printStackTrace();
             }
 
-
+            // Test all links in a Study Guide
             int response;
             for (String link : studyGuideLinks) {
-//                Document doc2 = (Document) Jsoup.connect(link).response();
                 try {
-                    Connection connection = Jsoup.connect(link).ignoreHttpErrors(true);
-
+                    Connection linkConnection = Jsoup.connect(link).ignoreHttpErrors(true);
 
-                    Document doc2 = connection.get();
-                    response = connection.response()
-                            .statusCode();
+                    // Although the `linkContents` is never utilized, it is a
+                    // required call. This is because Jsoup expects that `get()`
+                    // is called before the `response()` object is available.
+                    Document linkContents = linkConnection.get();
+                    response = linkConnection.response().statusCode();
 
-                    if (response >= 400) {
+                    // Report a bad link if anything other than a "200 OK" response is returned.
+                    if (response != 200) {
                         badLinks.add(link);
-                        writeToFile.createTxtFile(titleOfStudyGuide, link, link2);
```

Comment:
Previously there was a bunch of commented out code here, and you've pulled that into a new method, which is a good thing.